### ISSUE TITLE
setBothDrawables=true renders black background on API 19

### ISSUE THREAD

**Issue by rdadkins** [DreaminginCodeZH/MaterialProgressBar#62]
Here is my current configuration:
```
<me.zhanghai.android.materialprogressbar.MaterialProgressBar
        style="@style/Widget.MaterialProgressBar.ProgressBar.Horizontal.NoPadding"
        android:id="@+id/cardProgressBar"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:progress="10"
        android:secondaryProgress="20"
        android:max="100"
        app:mpb_progressStyle="horizontal"
        app:mpb_useIntrinsicPadding="false"
        app:mpb_setBothDrawables="true"
        app:mpb_showProgressBackground="true"/>
```
On API 19, it renders as follows:
![](http://i.imgur.com/0XNHzEg.png)

And on API 24, it renders as expected:
![](http://i.imgur.com/xDjyd53.png)

When just using either indeterminate or determinate mode, everything renders as expected. Only when `mpb_setBothDrawables` is `true`, it renders black on 19.

**Comment by rdadkins** [comment]
This actually seems to hold true for a regular horizontal progress bar on 19. If any tinting is applied , the background is completely black. Indeterminate bars render as expected.

**Comment by rdadkins** [comment]
It actually appears that the determinate tinted examples provided by the library are _over_ tinted. The third determinate progress bar's background color is a very dark red, not black. The second one, however, is black.

![](http://i.imgur.com/NbxU9fo.png)

**Comment by zhanghai** [comment]
Strange. Which ROM are you using? Can this be reproduced with an API 19 Android emulator?

> This actually seems to hold true for a regular horizontal progress bar on 19. If any tinting is applied , the background is completely black.

What do you mean by "regular"? Does framework `ProgressBar` work?

**Comment by rdadkins** [comment]
I meant "regular" as a library provided horizontal bar without tinting (i.e. the second in the sample project), not framework. The framework bar looks fine

The original statement `If any tinting is applied, the background is completely black` is now misleading. The secondary and background tinting seem to have a really high alpha applied to them which is noted by the third bar in the sample provided. It is not black, just a really dark red (same for the second, it is a really dark green). 

> Which ROM are you using?

YTP801A113300

> Can this be reproduced with an API 19 Android emulator?

It looks like it. I am just using a stock API 19 emulator with Google APIs and it looks just like above.

**Comment by zhanghai** [comment]
Bisected to cf059b0c24ba3519ac4b8b57930b1a5c599b6bb2 . Strange, still investigating.

**Comment by zhanghai** [comment]
Found out it's a bug of `ColorFilter` with alpha in color which is resulting in the black, and it can be workarounded by turning on software layer rendering. Released in v1.4.2.

**Comment by rdadkins** [comment]
Works as expected, thank you for updating!
